Sources/TokamakCore/Fiber/Layout/ViewSpacing.swift

```diff
@@ -0,0 +1,104 @@
+// Copyright 2022 Tokamak contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+//  Created by Carson Katri on 6/20/22.
+//
+
+import Foundation
+
+/// The preferred spacing around a `View`.
+///
+/// When computing spacing in a custom `Layout`, use `distance(to:along:)`
+/// to find the smallest spacing needed to accommodate the preferences
+/// of the `View`s you are aligning.
+public struct ViewSpacing {
+  /// The `View` type this `ViewSpacing` is for.
+  /// Some `View`s prefer different spacing based on the `View` they are adjacent to.
+  @_spi(TokamakCore)
+  public var viewType: Any.Type?
+  private var top: (ViewSpacing) -> CGFloat
```

Comment:
formatting readability nit
```suggestion
  /// The `View` type this `ViewSpacing` is for.
  /// Some `View`s prefer different spacing based on the `View` they are adjacent to.
  @_spi(TokamakCore)
  public var viewType: Any.Type?

  private var top: (ViewSpacing) -> CGFloat
```